File: 백준/Gold/5430. AC/AC.cc
#include <iostream>
#include <string>
#include <deque>
using namespace std;

void func() {
    string word, arrays; 
    int n;
    cin >> word >> n >> arrays;
    
    string numStr;
    deque<int> dq;
    
    // 배열 파싱
    for (char ch : arrays) {
        if (isdigit(ch)) {
            numStr += ch;
        }
        else if (!numStr.empty()) {
            dq.push_back(stoi(numStr));
            numStr.clear();
        }
    }
    
    bool isRev = false;
    
    // 명령어 처리
    for (char ch : word) {
        if (ch == 'R') {
            isRev = !isRev;
        }
        else if (ch == 'D') {
            if (dq.empty()) {
                cout << "error\n";
                return;
            }
            if (isRev)
                dq.pop_back();
            else
                dq.pop_front();
        }
    }
    
    // 결과 출력
    cout << '[';
    if (isRev) {
        // 역순으로 출력
        for (int i = dq.size() - 1; i >= 0; i--) {
            cout << dq[i];
            if (i > 0) cout << ',';
        }
    }
    else {
        // 정순으로 출력
        for (int i = 0; i < dq.size(); i++) {
            cout << dq[i];
            if (i < dq.size() - 1) cout << ',';
        }
    }
    cout << "]\n";
}

int main() {
    ios::sync_with_stdio(false); 
    cin.tie(nullptr); 
    cout.tie(nullptr);
    
    int T;
    cin >> T;
    while (T--) {
        func();
    }
    return 0;
}

File: 백준/Gold/5430. AC/AC.java
import java.util.*;
import java.io.*;


public class Main {

    public static void func(BufferedReader br) throws IOException {
        String word = br.readLine();
        int n = Integer.parseInt(br.readLine());
        String arrays = br.readLine();

        Deque<Integer> dq = new ArrayDeque<>();

        // 배열 파싱
        StringBuilder numStr = new StringBuilder();
        for (char ch : arrays.toCharArray()) {
            if (Character.isDigit(ch)) {
                numStr.append(ch);
            } else if (numStr.length() > 0) {
                dq.addLast(Integer.parseInt(numStr.toString()));
                numStr.setLength(0);
            }
        }

        boolean isRev = false;

        for (char ch : word.toCharArray()) {
            if (ch == 'R') {
                isRev = !isRev;
            } else if (ch == 'D') {
                if (dq.isEmpty()) {
                    System.out.println("error");
                    return;
                }
                if (isRev) dq.pollLast();
                else dq.pollFirst();
            }
        }

        // 결과 출력
        StringBuilder sb = new StringBuilder();
        sb.append('[');
        if (isRev) {
            Iterator<Integer> it = dq.descendingIterator();
            while (it.hasNext()) {
                sb.append(it.next());
                if (it.hasNext()) sb.append(',');
            }
        } else {
            Iterator<Integer> it = dq.iterator();
            while (it.hasNext()) {
                sb.append(it.next());
                if (it.hasNext()) sb.append(',');
            }
        }
        sb.append(']');
        System.out.println(sb);
    }

    public static void main(String[] args) throws IOException {
        // 빠른 입출력
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int T = Integer.parseInt(br.readLine());
        while (T-- > 0) {
            func(br);
        }
    }
}
